app/models/hud_reports/report_instance.rb

```diff
@@ -79,6 +90,19 @@ def failures
       report_cells.where.not(error_messages: nil).pluck(:question, :cell_name, :status, :error_messages)
     end
 
+    private def job_failed?
+      related_job.present? && related_job.failed?
+    end
+
+    def related_job
+      # See if we can find a related job (this is really overloading the jobs_for_class scope, but should work)
+      dj = Delayed::Job.jobs_for_class('RunReportJob').jobs_for_class(id.to_s)
+      return dj&.first if dj.count == 1
+
+      # If we didn't find an obvious match, just return nothing
+      nil
```

Comment:
Not sure the order is necessary since we explicitly only return it if there is one, but I like the simplification, thanks!

Reply:
oh yeah, I have the habit of always using `first` and `order` together since it ensures behavior is deterministic.

In this case there are two queries executed on this line (many and first). So you could have a super tiny edge case with `many` returning false, a new job is added, and then `first` returns one of those two records. So always returning the same record is helpful for consistency